library/core/src/main/java/com/google/android/exoplayer2/source/SampleDataQueue.java

```diff
@@ -265,8 +270,10 @@ private void readEncryptionData(DecoderInputBuffer buffer, SampleExtrasHolder ex
         encryptedDataSizes[i] = scratch.readUnsignedIntToInt();
       }
     } else {
-      clearDataSizes[0] = 0;
-      encryptedDataSizes[0] = extrasHolder.size - (int) (offset - extrasHolder.offset);
+      int addedHeaderSize = MimeTypes.AUDIO_AC4.equals(mimeType) ? 7 : 0;
+      clearDataSizes[0] = addedHeaderSize;
```

Comment:
See comment elsewhere about not modifying this class at all, however as an aside I don't think you should be assuming there's no sub-sample encryption (i.e. you'd also need to do something in the `if` block above, as well as the `else` block here).

Reply:
@ojw28 The fact is I added code in `if` block but I removed it when I submitted this pull request. The reason is there is no this kind of stream inside Dolby so that I can't test that `if` branch. I can double check with our content creation team whether they will create this kind of stream in future. BTW, according to your experience, is there sub-sample encryption content for audio track (e.g. aac) in practice? Thanks.

Reply:
I'm not sure whether sub-sample encryption is really a thing for audio samples, but I think we should handle the possibility :).